### DISCUSSION TITLE
Use atom within hooks

### DISCUSSION THREAD

**Discussion by anthony-bernardo** [pmndrs/jotai#2838] [Q&A]
How to get value of an atom outside react, but in a hook ?

When I try to update an atom in a hook, and the retrieve the update atom, I'm not getting the updated atom

`const ui = useAtomValue(UIList)`

**Comment by dai-shi** [comment]
Do you mean you want to get an updated atom value in a callback?

**Reply by anthony-bernardo** [reply]
Sorry @dai-shi , I'm in a react context. It's a sort of hooks yes. It drive me crazy, I cannot retrieve the value in the hook, I have to done this to make this working : 

```
export const useDisplayFinalQte = () => {
    const setUI = useSetAtom(UIList)

    const execute = () => {
        const unsub = jotaiStore.sub(UIList, () => {
            unsub()
           
            // ...rest of the code
        })
    }

    return {
        execute,
    }

```

```
export const OnAfterLoad = () => {
    const displayFinalQte = useDisplayFinalQte()

    useEffect(() => {
        displayFinalQte.execute()
    }, [])

    return null
}
```

```
export const SalesDetailPage = () => {
    return (
        <Provider store={jotaiStore}>
            <Page />
        </Provider>
    )
}
```

**Reply by dai-shi** [reply]
> I cannot retrieve the value in the hook,

Which hook is it?

A full minimal reproduction with https://stackblitz.com would be helpful.

**Reply by anthony-bernardo** [reply]
I will make a reproduction soon

And it's a custom hook

https://react.dev/learn/reusing-logic-with-custom-hooks

**Comment by dmaskasky** [comment]
What was the resolution to this question?
